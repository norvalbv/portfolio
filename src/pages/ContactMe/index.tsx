import React, { ReactElement, useState } from 'react';
import CardWrapper from 'components/CardWrapper';
import TextField from 'components/TextField';
import ContactDetails from 'components/ContactDetails';

const ContactMe = (): ReactElement => {
  const [query, setQuery] = useState({
    name: '',
    email: '',
    message: '',
  });

  const setQueryByKey = (key: keyof typeof query, value: string | boolean): void => {
    setQuery({ ...query, [key]: value });
  };

  return (
    <CardWrapper title="Come Say Hi!" className="top-[2rem] w-10/12 min-w-[20rem] lg:top-[4rem]">
      <div className="flex flex-col gap-6 lg:flex-row">
        <form
          action="https://formsubmit.co/benjinorval@gmail.com"
          method="POST"
          className="rounded-lg bg-white/70 p-6 dark:bg-dark-dark/30 lg:w-3/5"
        >
          <h2 className="mb-6 underline underline-offset-8">Contact Form:</h2>
          <TextField
            label="Name"
            value={query.name}
            onChange={(val): void => setQueryByKey('name', val)}
            required
          />
          <TextField
            label="Email"
            value={query.email}
            onChange={(val): void => setQueryByKey('email', val)}
            required
            className="my-6"
          />
          <TextField
            label="Message"
            value={query.message}
            onChange={(val): void => setQueryByKey('message', val)}
            inputType="textarea"
            required
          />
          <button type="submit" className="mt-6 hover:underline">
            Submit
          </button>
        </form>
        <ContactDetails />
      </div>
    </CardWrapper>
  );
};

export default ContactMe;
