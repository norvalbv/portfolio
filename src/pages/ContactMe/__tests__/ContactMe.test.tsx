import React from 'react';
import renderer from 'react-test-renderer';

import ContactMe from '..';

describe('<ContactMe />', () => {
  test('should render correctly', () => {
    const tree = renderer.create(<ContactMe />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
