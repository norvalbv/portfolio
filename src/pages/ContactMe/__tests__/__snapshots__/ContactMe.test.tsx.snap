// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactMe /> should render correctly 1`] = `
<div
  className="relative mx-auto flex-1 mt-6 w-10/12 min-w-[20rem] md:mt-10"
>
  <div>
    <header
      className="flex items-center justify-between"
    >
      <section
        className="flex w-full flex-col gap-1"
      >
        <h1
          className="mb-[4.5rem] w-max border-b pb-2 text-5xl font-semibold md:text-6xl"
        >
          H
          e
          l
          l
          o
           
          W
          o
          r
          l
          d
          !
        </h1>
      </section>
    </header>
    <div
      className="flex flex-col gap-6 lg:flex-row"
    >
      <form
        className="rounded-lg bg-white/70 p-6 dark:bg-dark-dark/30 lg:w-3/5"
        onSubmit={[Function]}
      >
        <h2
          className="mb-6 underline underline-offset-8"
        >
          Contact Form:
        </h2>
        <fieldset>
          <div
            className="mb-2 flex justify-between"
          >
            <div
              className="text-light-text dark:text-dark-text"
            >
              <label
                className="flex"
              >
                Name
                <span
                  className="ml-1 text-xs text-accent-secondary"
                >
                  *
                </span>
              </label>
            </div>
          </div>
          <div>
            <div
              className="flex items-center"
            >
              <input
                autoComplete="on"
                className="appearance-none rounded focus:text-accent-secondary focus:outline focus:outline-light-text focus:ring-transparent dark:focus:outline-dark-text borber-b h-12 w-full bg-dark-dark/20 p-4 transition-[height] duration-300 dark:bg-white/20"
                name=""
                onChange={[Function]}
                placeholder=""
                required={true}
                style={
                  {
                    "fontSize": "inherit",
                  }
                }
                type="text"
                value=""
              />
            </div>
          </div>
        </fieldset>
        <fieldset
          className="my-6"
        >
          <div
            className="mb-2 flex justify-between"
          >
            <div
              className="text-light-text dark:text-dark-text"
            >
              <label
                className="flex"
              >
                Email
                <span
                  className="ml-1 text-xs text-accent-secondary"
                >
                  *
                </span>
              </label>
            </div>
          </div>
          <div>
            <div
              className="flex items-center"
            >
              <input
                autoComplete="on"
                className="appearance-none rounded focus:text-accent-secondary focus:outline focus:outline-light-text focus:ring-transparent dark:focus:outline-dark-text borber-b h-12 w-full bg-dark-dark/20 p-4 transition-[height] duration-300 dark:bg-white/20"
                name=""
                onChange={[Function]}
                placeholder=""
                required={true}
                style={
                  {
                    "fontSize": "inherit",
                  }
                }
                type="email"
                value=""
              />
            </div>
          </div>
        </fieldset>
        <fieldset>
          <div
            className="mb-2 flex justify-between"
          >
            <div
              className="text-light-text dark:text-dark-text"
            >
              <label
                className="flex"
              >
                Message
                <span
                  className="ml-1 text-xs text-accent-secondary"
                >
                  *
                </span>
              </label>
            </div>
          </div>
          <div>
            <textarea
              autoComplete="on"
              className="appearance-none rounded focus:text-accent-secondary focus:outline focus:outline-light-text focus:ring-transparent dark:focus:outline-dark-text borber-b w-full bg-dark-dark/20 p-4 transition-[height] duration-300 dark:bg-white/20 h-12"
              name=""
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              placeholder=""
              required={true}
              value=""
            />
          </div>
        </fieldset>
        <button
          className="mt-6 hover:underline"
          type="submit"
        >
          Submit
        </button>
      </form>
      <div
        className="flex flex-col gap-2 rounded-lg bg-white/70 p-6 dark:bg-dark-dark/30 lg:w-2/5"
      >
        <h2
          className="mb-6 underline underline-offset-8"
        >
          Contact Details:
        </h2>
        <p
          className="mb-4"
        >
          If you would like to contact me directly;
        </p>
        <span
          className="my-1 text-sm md:text-base"
        >
          Email
          : 
          <a
            className="my-4 underline transition duration-150 ease-linear hover:text-accent-secondary"
            download={false}
            href="mailto:benjinorval@gmail.com"
            rel="noreferrer"
            target="_blank"
          >
            benjinorval@gmail.com
          </a>
        </span>
        <span
          className="my-1 text-sm md:text-base"
        >
          Phone
          : 
          <a
            className="my-4 underline transition duration-150 ease-linear hover:text-accent-secondary"
            download={false}
            href="tel:+447508719043"
            rel="noreferrer"
            target="_blank"
          >
            (+44) 7508719043
          </a>
        </span>
        <span
          className="my-1 text-sm md:text-base"
        >
          Linked In
          : 
          <a
            className="my-4 underline transition duration-150 ease-linear hover:text-accent-secondary"
            download={false}
            href="https://www.linkedin.com/in/benjamin-norval/"
            rel="noreferrer"
            target="_blank"
          >
            Benjamin Norval
          </a>
        </span>
        <span
          className="my-1 text-sm md:text-base"
        >
          GitHub
          : 
          <a
            className="my-4 underline transition duration-150 ease-linear hover:text-accent-secondary"
            download={false}
            href="https://github.com/norvalbv/"
            rel="noreferrer"
            target="_blank"
          >
            norvalbv
          </a>
        </span>
        <span
          className="my-1 text-sm md:text-base"
        >
          My CV
          : 
          <a
            className="my-4 underline transition duration-150 ease-linear hover:text-accent-secondary"
            download={false}
            href="test-file-stub"
            rel="noreferrer"
            target="_blank"
          >
            Download Here
          </a>
        </span>
        <p
          className="mt-6 text-sm"
        >
          I look forward to hearing from you!
        </p>
      </div>
    </div>
  </div>
</div>
`;
